application.go

```diff
@@ -0,0 +1,128 @@
+package civogo
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Application is the struct for the Application model
+type Application struct {
+	Name        string        `json:"name" validate:"required" schema:"name"`
+	AccountID   string        `json:"account_id"`
+	ID          string        `json:"id"`
+	NetworkID   string        `json:"network_id" validate:"required" schema:"network_id"`
+	Description string        `json:"description"`
+	ProcessInfo []ProcessInfo `json:"process_info,omitempty"`
+	Domains     []string      `json:"domains,omitempty"`
+	SSHKeyIDs   []string      `json:"ssh_key_ids,omitempty"`
+	Config      []EnvVar      `json:"config,omitempty"`
+	// Status can be one of:
+	// - "building":  Implies cluster is building
+	// - "available": Implies cluster is ready
+	// - "ready": Implies app is ready
+	Status string `json:"status"`
+}
+
+// EnvVar holds key-value pairs for an application
+type EnvVar struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+// ProcessInfo contains the information about the process obtained from Procfile
+type ProcessInfo struct {
+	ProcessType  string `json:"process_type"`
+	ProcessCount int    `json:"process_count"`
+}
+
+// ListApplications returns all applications in the account
```

Comment:
Account or region? As these are stored in Kubernetes, I think it's region specific